kong/runloop/handler.lua

```diff
@@ -1438,7 +1438,7 @@ return {
 
       local upstream_status_header = constants.HEADERS.UPSTREAM_STATUS
       if kong.configuration.enabled_headers[upstream_status_header] then
-        local upstream_status = ctx.buffered_status or tonumber(sub(var.upstream_status or "", -3))
+        local upstream_status = ctx.buffered_status or tonumber(sub(var.upstream_status or "", -3)) or ngx.status
```

Comment:
What if Kong shortcircuits the request, aka the answer does not come from the upstream? It feels like it is not OK to always fallback to `ngx.status` on this.

Reply:
@bungle If the request doesn't reach out the upstream, it means the req is not proxied (In the `before` phase of header_filter in runloop.handler, there is a quick `return`), then it won't have the chance to inject the `UPSTREAM_STATUS` header.
In `proxy-cache`, it actually sets `ngx.status` inside `kong.response.exit(res.status, ..., ...)`, that' why I use `ngx.status` as the last alternative.